crates/commonware-node/src/consensus/application/executor.rs

```diff
@@ -325,22 +335,15 @@ where
         block: Block,
         acknowledgment: Exact,
     ) -> eyre::Result<()> {
-        if let Err(error) = self
-            .canonicalize(
-                Span::current(),
-                HeadOrFinalized::Finalized,
-                block.height(),
-                block.digest(),
-                oneshot::channel().0,
-            )
-            .await
-        {
-            warn!(
-                %error,
-                "failed canonicalizing finalized block; will still attempt \
-                forwarding it to the execution layer",
-            );
-        }
+        self.canonicalize(
+            Span::current(),
+            HeadOrFinalized::Finalized,
+            block.height(),
+            block.digest(),
+            oneshot::channel().0,
+        )
+        .await
+        .wrap_err("failed canonicalizing finalized block")?;
```

Comment:
Probably fine to just exit here considering we can't handle this. Finalizations must go through.